src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -46,13 +46,72 @@ class ActivityFeed extends React.Component<Props, State> {
 
     feedContainer: null | HTMLElement;
 
+    componentDidMount() {
+        this.scrollFeedContainerToBottom(false);
+    }
+
+    componentDidUpdate(prevProps: Props, prevState: State) {
+        this.scrollFeedContainerToBottom(true, prevProps, prevState);
+    }
+
+    /**
+     * Determines when the sidebar should scroll to the bottom
+     * @param {boolean} componentWasUpdated - True if this call came from componentDidUpdate
+     * @param {object} prevProps - Exists only if this call came from componentDidUpdate
+     * @param {object} prevState - Exists only if this call came from componentDidUpdate
+     */
+    scrollFeedContainerToBottom = (
+        componentWasUpdated: boolean,
+        prevProps?: Props,
+        prevState?: State,
+    ) => {
+        if (this.feedContainer) {
+            if (componentWasUpdated) {
```

Comment:
This introduction of this boolean is adding a lot of complexity, since it requires existence checks on `prevProps` and `prevState`. Instead, focusing on reusing the common logic, which checks `feedContainer` and resets the `scrollTop` attribute.

```
    componentDidMount() {
        this.resetFeedScroll();
    }

    componentDidUpdate(prevProps, prevState: State) {
        const {
            currentUser: previousUser,
            feedItems: prevFeedItems,
        } = prevProps;
        const { currentUser, feedItems: currFeedItems } = this.props;
        const hasFewerItems = prevFeedItems && currFeedItems && prevFeedItems.length < currFeedItems.length;
        const hasNewItems = !prevFeedItems && currFeedItems;
        const hasNewUser = currentUser !== previousUser;

        if (hasFewerItems || hasNewItems || hasNewUser) {
            this.resetFeedScroll();
        }

        // TODO: Handle state changes similarly
    }

    resetFeedScroll() {
        if (this.feedContainer) {
            this.feedContainer.scrollTop = this.feedContainer.scrollHeight;
        }
    }
```